src/psyclone/psyGen.py

```diff
@@ -3773,6 +3812,38 @@ def get_kernel_schedule(self):
             self._kern_schedule = astp.generate_schedule(self.name, self.ast)
         return self._kern_schedule
 
+    def set_opencl_options(self, options):
+        '''
+        Validate and store a set of options associated to the Kernel to
+        tune the OpenCL code generation.
+
+        :param options: a set of options to tune the OpenCL code.
+        :type options: map of <string>:<value>
+
+        '''
+        # Validate that the options given are supported
+        for key, value in options.items():
+            if key == "local_size":
+                if not isinstance(value, int):
+                    raise TypeError(
+                        "CodedKern opencl_option 'local_size' should be an "
+                        "integer.")
+            elif key == "queue_number":
+                if not isinstance(value, int):
+                    raise TypeError(
+                        "CodedKern opencl_option 'queue_number' should be an "
+                        "integer.")
+            else:
+                raise AttributeError(
+                    "CodedKern does not support the opencl_option '{0}'."
+                    "".format(key))
```

Comment:
Please include the valid options in this error message.

Reply:
Done